src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectSnapshotManagerBase.cs

```diff
@@ -9,20 +9,28 @@ internal abstract class ProjectSnapshotManagerBase : ProjectSnapshotManager
     {
         public abstract Workspace Workspace { get; }
 
-        public abstract void ProjectAdded(Project underlyingProject);
+        public abstract void ProjectUpdated(ProjectSnapshotUpdateContext update);
 
-        public abstract void ProjectChanged(Project underlyingProject);
+        public abstract void HostProjectAdded(HostProject hostProject);
 
-        public abstract void ProjectUpdated(ProjectSnapshotUpdateContext update);
+        public abstract void HostProjectChanged(HostProject hostProject);
+
+        public abstract void HostProjectRemoved(HostProject hostProject);
 
-        public abstract void ProjectRemoved(Project underlyingProject);
+        public abstract void HostProjectBuildComplete(HostProject hostProject);
```

Comment:
Changed `HostProjectBuildComplete(HostProject hostProject)` to take a `HostProject` because it makes more sense :wink: